ee/server/service/hosts.go

```diff
@@ -190,6 +194,95 @@ func (svc *Service) UnlockHost(ctx context.Context, hostID uint) (string, error)
+		return err
+	}
+
+	// wipe validations are based on the platform of the host, Windows and macOS
+	// require MDM to be enabled and the host to be MDM-enrolled in Fleet. Linux
+	// uses scripts, not MDM.
+	var requireMDM bool
+	switch host.FleetPlatform() {
+	case "darwin":
+		if err := svc.VerifyMDMAppleConfigured(ctx); err != nil {
+			if errors.Is(err, fleet.ErrMDMNotConfigured) {
+				err = fleet.NewInvalidArgumentError("host_id", fleet.AppleMDMNotConfiguredMessage).WithStatus(http.StatusBadRequest)
+			}
+			return ctxerr.Wrap(ctx, err, "check macOS MDM enabled")
+		}
+		requireMDM = true
+
+	case "windows":
+		if err := svc.VerifyMDMWindowsConfigured(ctx); err != nil {
+			if errors.Is(err, fleet.ErrMDMNotConfigured) {
+				err = fleet.NewInvalidArgumentError("host_id", fleet.WindowsMDMNotConfiguredMessage).WithStatus(http.StatusBadRequest)
+			}
+			return ctxerr.Wrap(ctx, err, "check windows MDM enabled")
+		}
+		requireMDM = true
+
+	case "linux":
+		// on linux, a script is used to wipe the host so scripts must be enabled
+		appCfg, err := svc.ds.AppConfig(ctx)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "get app config")
+		}
+		if appCfg.ServerSettings.ScriptsDisabled {
+			return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("host_id", "Can't wipe host because running scripts is disabled in organization settings."))
+		}
+
+	default:
+		return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("host_id", fmt.Sprintf("Unsupported host platform: %s", host.Platform)))
+	}
+
+	if requireMDM {
+		// the wipe command requires the host to be MDM-enrolled in Fleet
+		hostMDM, err := svc.ds.GetHostMDM(ctx, host.ID)
+		if err != nil {
+			if fleet.IsNotFound(err) {
+				return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("host_id", "Can't wipe the host because it doesn't have MDM turned on."))
+			}
+			return ctxerr.Wrap(ctx, err, "get host MDM information")
+		}
+		if !hostMDM.IsFleetEnrolled() {
+			return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("host_id", "Can't wipe the host because it doesn't have MDM turned on."))
+		}
+	}
+
+	// validations based on host's actions status (pending lock, unlock, wipe)
+	lockWipe, err := svc.ds.GetHostLockWipeStatus(ctx, host)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "get host lock/wipe status")
+	}
+	switch {
+	case lockWipe.IsPendingLock():
+		return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("host_id", "Host has pending lock request. Host cannot be wiped until lock is complete."))
+	case lockWipe.IsPendingUnlock():
+		return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("host_id", "Host has pending unlock request. Host cannot be wiped until unlock is complete."))
+	case lockWipe.IsPendingWipe():
+		return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("host_id", "Host has pending wipe request. The host will be wiped when it comes online."))
+	case lockWipe.IsWiped():
```

Comment:
@mna was adding some CLI tests for wipe and I realised there were no cases to check if a host is wiped before unlocking/locking the host. Should we add a check to see if the host is wiped first before doing these commands like we do here?

Reply:
ah ok i just saw the [comment here](https://github.com/fleetdm/fleet/issues/9951#issuecomment-1955108805). for now i will leave those cases out of the lock/unlock tests.

Reply:
@ghernandez345 that's a good question, I don't know if technically this could happen because a wiped host would need to be re-enrolled and would be a different host (i.e. fleetd would have been wiped, I presume? Although I don't know 100% if that would come up as a different host, given that the serial number would be the same). It's probably good to add this IsWiped check to prevent lock/unlock, but then again... a wiped host would always remain in `wiped` state (we have no hook at the moment that transitions a host from `wiped` to the default `unlocked` AFAIK?).

I'll leave a question for Noah on the ticket, I'm not sure how that's supposed to be handled.

Reply:
https://github.com/fleetdm/fleet/issues/9951#issuecomment-1964438980